pkg/controller/reconciler.go

```diff
@@ -68,19 +68,35 @@ func (o *operationAnnotationWrapper) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
+	if ok, err := o.removeOperationAnnotation(obj, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile, v1beta1constants.GardenerOperationRestore); !ok {
+		return reconcile.Result{}, err
+	}
+
+	return o.Reconciler.Reconcile(request)
+}
+
+func (o *operationAnnotationWrapper) removeOperationAnnotation(obj runtime.Object, key string, values ...string) (bool, error) {
+	var opAnnotationRemoved bool
 	acc, err := meta.Accessor(obj)
 	if err != nil {
-		return reconcile.Result{}, err
+		return false, err
 	}
 
 	annotations := acc.GetAnnotations()
-	if annotations[v1beta1constants.GardenerOperation] == v1beta1constants.GardenerOperationReconcile {
-		withOpAnnotation := obj.DeepCopyObject()
-		delete(annotations, v1beta1constants.GardenerOperation)
-		acc.SetAnnotations(annotations)
-		if err := o.client.Patch(o.ctx, obj, client.MergeFrom(withOpAnnotation)); err != nil {
-			return reconcile.Result{}, err
+	objWithOpAnnotation := obj.DeepCopyObject()
+	for _, value := range values {
+		if annotations[key] == value {
+			delete(annotations, key)
+			acc.SetAnnotations(annotations)
+			opAnnotationRemoved = true
```

Comment:
nit: you can return from the loop here: 
move the patch command here and then you can get rid of `opAnnotationRemoved`

Reply:
You are right.